dynamodb-enhanced/runtime/src/main/java/io/quarkus/amazon/dynamodb/enhanced/runtime/DynamodbEnhancedClientRecorder.java

```diff
@@ -28,10 +43,32 @@ public BeanContainerListener setDynamoDbAsyncClient() {
             @Override
             public void created(BeanContainer container) {
                 DynamodbEnhancedClientProducer producer = container.beanInstance(DynamodbEnhancedClientProducer.class);
-                producer.setDynamoDbAsyncClient(container.beanInstance(DynamoDbAsyncClient.class));
+                producer.setDynamoDbAsyncClient(container.beanInstance(DynamoDbAsyncClient.class),
+                        createExtensionList());
             }
         };
 
         return beanContainerListener;
     }
+
+    private List<DynamoDbEnhancedClientExtension> createExtensionList() {
+        if (this.buildTimeConfig.clientExtensions.isPresent() && !this.buildTimeConfig.clientExtensions.get().isEmpty()) {
+            return this.buildTimeConfig.clientExtensions.get().stream().map(configValue -> {
+                DynamoDbEnhancedClientExtension extension = ChainExtension.create();
+                switch (configValue) {
+                    case AtomicCounterExtension:
+                        extension = AtomicCounterExtension.builder().build();
+                        break;
+                    case VersionedRecordExtension:
+                        extension = VersionedRecordExtension.builder().build();
+                        break;
+                    case AutoGeneratedTimestampRecordExtension:
+                        extension = AutoGeneratedTimestampRecordExtension.create();
+                        break;
+                }
+                return extension;
+            }).collect(Collectors.toList());
```

Comment:
We try to avoid lambdas in runtime code because they have a cost.

Reply:
Done.